virttest/qemu_vm.py

```diff
@@ -1724,6 +1770,18 @@ def sort_key(dev):
                 raise virt_vm.VMDeviceError("The numa node cfg can not fit"
                                             " smp and memory cfg.")
 
+        # Add '-numa hmat-lb' and '-numa hmat-cache'
+        if devices.has_option('numa hmat-lb,.*'):
+            for numa_node in params.objects('guest_numa_nodes'):
+                numa_params = params.object_params(numa_node)
+                nodeid = numa_params['numa_nodeid']
+                initiator = numa_params.get('numa_initiator', nodeid)
+                for hmat_lb in numa_params.objects('numa_hmat_lb'):
+                    hmat_lb_params = params.object_params(hmat_lb)
+                    cmd = add_numa_hmat_lb(initiator, nodeid, hmat_lb_params)
+                    devices.insert(StrDev('numa_hmat_lb', cmdline=cmd))
+                devices.insert(add_numa_hmat_cache(nodeid))
```

Comment:
I'd like to describe the relationship between hmat-cache and hmat-lb.
When add hmat-cache, both the latency and bandwidth need to be configured in hmat-lb. So for one hmat-cache, at least two hmat-lb are needed.
However, -numa hmat-lb does not always come with hmat-cache. It's possible to define a hmat-lb without hmat-cache.

Reply:
@huangyum Thanks for the information, I will make modification accordingly.

Reply:
@huangyum But, what do you expect when adding the devices? Cause from the command line their relationship is only they all restrict to the same nodeid, all the other parameters still need to be defined by the user anyway, can not be generated automatically since there is no proper rule for it(like default value to be set). Here seems we can only check if there are suitable `hmat_lb` devices been defined when add `hmat_cache`, do you think that's enough? Please help correct me if I was wrong. Thanks.

Reply:
Yes, it should be fine. Just not every node has a hmat_cache, so I guess we might need add check before add hmat-cache.